### PR TITLE
General Availability version of PSTN dial-out capability feature

### PR DESCRIPTION
**Pull request by paolamvhz**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/communication/azure-communication-rooms/azure/communication/rooms/_generated/aio/_client.py [hunk 1]
```diff
@@ -34,15 +35,34 @@ def __init__(  # pylint: disable=missing-client-constructor-parameter-credential
     ) -> None:
         _endpoint = "{endpoint}"
         self._config = AzureCommunicationRoomsServiceConfiguration(endpoint=endpoint, **kwargs)
-        self._client: AsyncPipelineClient = AsyncPipelineClient(base_url=_endpoint, config=self._config, **kwargs)
+        _policies = kwargs.pop("policies", None)
+        if _policies is None:
```
Comment (mikehang-msft) [inline sdk/communication/azure-communication-rooms/azure/communication/rooms/_generated/aio/_client.py:39]: I believe this is a new generated codes. I do not think it will affect the changes but to ensure, please run all tests and sample scenarios.

**Comment by mikehang-msft** [comment]
Can you please update ChangeLog.md

File: sdk/communication/azure-communication-rooms/CHANGELOG.md [hunk 2]
```diff
@@ -1,16 +1,7 @@
...
-- Added support for a new communication identifier `MicrosoftTeamsAppIdentifier`.
-
-### Breaking Changes
-
-### Bugs Fixed
-
-### Other Changes
-- The `MicrosoftBotIdentifier` and `MicrosoftBotProperties` have been deprecated.
+## 1.1.0 (2024-04-15)
+General Availability version of PSTN dial-out capability feature
```
Comment (mikehang-msft) [inline sdk/communication/azure-communication-rooms/CHANGELOG.md:4]: Nit: please add a dot at the end. "General Availability version of PSTN dial-out capability feature.". I am going to make changes to .NET SDK as well.

**Review by mikehang-msft — APPROVED** [review]

File: sdk/communication/azure-communication-rooms/CHANGELOG.md (hunk 2 above)
Comment (paolamvhz) [reply sdk/communication/azure-communication-rooms/CHANGELOG.md]: done thanks

**Comment by mikehang-msft** [comment]
@paolamvhz, it appears the following file needs to be updated to use the latest version (1.1.0). Can you please update? 
https://github.com/paolamvhz/azure-sdk-for-python/blob/roomsPSTGA/sdk/communication/azure-communication-rooms/azure/communication/rooms/_api_versions.py

**Comment by mikehang-msft** [comment]
Please update this file as well.

https://github.com/Azure/azure-sdk-for-python/blob/d95e01d580e3c0d0085bf0dcb71776f3e5997b7f/sdk/communication/azure-communication-rooms/azure/communication/rooms/_rooms_client.py

**Comment by paolamvhz** [comment]
> Please update this file as well
> 
> https://github.com/Azure/azure-sdk-for-python/blob/d95e01d580e3c0d0085bf0dcb71776f3e5997b7f/sdk/communication/azure-communication-rooms/azure/communication/rooms/_rooms_client.py

fixed

**Review by mikehang-msft — APPROVED** [review]

File: sdk/communication/azure-communication-rooms/setup.py [hunk 3]
```diff
@@ -65,7 +65,7 @@
         ]
     ),
     install_requires=[
-        "azure-core>=1.24.0",
+        "azure-core>=1.29.0",
         "msrest>=0.7.1",
```
Comment (annatisch) [inline sdk/communication/azure-communication-rooms/setup.py:69]: msrest should not be a required dependency, please update:
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/communication/azure-communication-chat/setup.py#L66-L69

File: sdk/communication/azure-communication-rooms/CHANGELOG.md [hunk 4]
```diff
@@ -1,16 +1,9 @@
 # Release History
 
-## 1.1.0b2 (Unreleased)
 
-### Features Added
-- Added support for a new communication identifier `MicrosoftTeamsAppIdentifier`.
```
Comment (annatisch) [inline sdk/communication/azure-communication-rooms/CHANGELOG.md:6]: Why did you remove the release notes?

Comment (mikehang-msft) [reply sdk/communication/azure-communication-rooms/CHANGELOG.md:6]: MicrosoftTeamsAppIdentifier feature has already been released. But not aware they added release note to rooms changeslog. In this case, I think we can keep the Features Added note and just replace the ## 1.1.0b2 (Unreleased) with out release version and description.

Comment (paolamvhz) [reply sdk/communication/azure-communication-rooms/CHANGELOG.md:6]: Updated

File: sdk/communication/azure-communication-rooms/setup.py [hunk 5]
```diff
@@ -65,8 +65,7 @@
         ]
     ),
     install_requires=[
-        "azure-core>=1.24.0",
-        "msrest>=0.7.1",
+        "azure-core>=1.29.0",
```
Comment (annatisch) [inline sdk/communication/azure-communication-rooms/setup.py:68]: This must include isodate

File: sdk/communication/azure-communication-rooms/CHANGELOG.md (hunk 4 above)
Comment (paolamvhz) [reply sdk/communication/azure-communication-rooms/CHANGELOG.md:6]: I removed the release notes from the communication identifier as they seem to be unreleased and cause the conflicts. The change in the changelog file was introduced here https://github.com/Azure/azure-sdk-for-python/commit/33608ac52f65386d0ba923446e8664c41e32c2a1#diff-998a1fc3d0f56f4f5e35b2e6099c151bffc6db6fdba8cbf411db895525d3ba40 but is not needed in rooms changeLog

File: sdk/communication/azure-communication-rooms/CHANGELOG.md [hunk 6]
```diff
@@ -12,6 +12,9 @@
 ### Other Changes
 - The `MicrosoftBotIdentifier` and `MicrosoftBotProperties` have been deprecated.
 
+## 1.1.0 (2024-04-15)
```
Comment (kristapratico) [inline sdk/communication/azure-communication-rooms/CHANGELOG.md:15]: What is the 1.1.0b2 version above? Should we move those entries to this version?

**Review by annatisch — APPROVED** [review]

**Review by mikehang-msft — APPROVED** [review]
